src/fontra/views/editor/editor.js

```diff
@@ -463,177 +454,53 @@ export class EditorController {
         sidebarContainer.classList.add("animating");
       }
     }, 100);
+  }
 
-    for (const sidebarTab of document.querySelectorAll(".sidebar-tab")) {
-      sidebarTab.addEventListener("click", (event) => {
-        this.toggleSidebar(sidebarTab.dataset.sidebarName, true);
-      });
-    }
+  addSidebar(sidebar) {
+    const editorContainer = document.querySelector(".editor-container");
+    sidebar.attach(editorContainer);
+    this.sidebars.push(sidebar);
+  }
 
-    this.initSidebarTextEntry();
-    this.initSidebarGutters();
+  addSidebarPanel(panelInstance, sidebarName) {
+    const sidebar = this.sidebars.find((sidebar) => sidebar.identifier === sidebarName);
+    sidebar.addPanel(panelInstance);
+    panelInstance.attach(this);
```

Comment:
I would prefer if the EditorController instance was passed to the constructor.

Reply:
this should be ok